servers/graphql-kotlin-spring-server/src/main/kotlin/com/expediagroup/graphql/server/spring/subscriptions/ApolloSubscriptionProtocolHandler.kt

```diff
@@ -149,15 +149,19 @@ class ApolloSubscriptionProtocolHandler(
     }
 
     private fun onInit(operationMessage: SubscriptionOperationMessage, session: WebSocketSession): Flux<SubscriptionOperationMessage> {
-        val connectionParams = getConnectionParams(operationMessage.payload)
-        val graphQLContext = contextFactory.generateContext(session) ?: DefaultGraphQLContext()
-        val onConnect = subscriptionHooks.onConnect(connectionParams, session, graphQLContext)
-        sessionState.saveContext(session, onConnect)
-        val acknowledgeMessage = Flux.just(acknowledgeMessage)
+        saveContext(operationMessage, session)
+        val acknowledgeMessage = Mono.just(acknowledgeMessage)
         val keepAliveFlux = getKeepAliveFlux(session)
         return acknowledgeMessage.concatWith(keepAliveFlux)
     }
 
+    private fun saveContext(operationMessage: SubscriptionOperationMessage, session: WebSocketSession) {
+        val connectionParams = getConnectionParams(operationMessage.payload)
+        val context = runBlocking { contextFactory.generateContext(session) }
```

Comment:
This line here is still ambiguous to me. If we wrapping this in `mono { }` then the code is not run unless the client subscribes and then it might not run first before the operations are exectuted